ext/hook/uhook.c

```diff
@@ -5,32 +5,29 @@
 
 #include "../compatibility.h"
 #include "../configuration.h"
+#include "../logging.h"
 
 #include "uhook_arginfo.h"
 
 #include <hook/hook.h>
 
 #include "uhook.h"
+#include "../ddtrace.h"
 
-extern void (*profiling_interrupt_function)(zend_execute_data *);
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
 
-static inline zval *ddtrace_hookdata_property_id(zend_object *hookdata) {
-    return OBJ_PROP_NUM(hookdata, 0);
-}
-static inline zval *ddtrace_hookdata_property_args(zend_object *hookdata) {
-    return OBJ_PROP_NUM(hookdata, 1);
-}
-static inline zval *ddtrace_hookdata_property_returned(zend_object *hookdata) {
-    return OBJ_PROP_NUM(hookdata, 2);
-}
-static inline zval *ddtrace_hookdata_property_exception(zend_object *hookdata) {
-    return OBJ_PROP_NUM(hookdata, 3);
-}
+extern void (*profiling_interrupt_function)(zend_execute_data *);
 
 zend_class_entry *ddtrace_hook_data_ce;
 
+static __thread HashTable dd_closure_hooks;
 static __thread HashTable dd_active_hooks;
 
+typedef struct {
+    size_t size;
+    zend_long id[1]; // struct hack
```

Comment:
Since this exists in a `.c` file and not a header file, I would prefer the standards-compliant flexible array member instead.

```suggestion
    zend_long id[];
```

Obviously other changes are required as well to do the math right.